### PR TITLE
Fix nodejs type mismatch

### PR DESCRIPTION
**Pull request by iwahbe**
There was a mismatch between `@types/node` and `typescript` on their
default versions. This was fixed by listing both of them explicitly as a
`devDependency`.

# Description

Fixes # (issue)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/codegen/nodejs/gen_program_test.go [hunk 1]
```diff
@@ -29,41 +30,51 @@ func nodejsCheck(t *testing.T, path string, dependencies codegen.StringSet) {
...
+	packageJSONPath := filepath.Join(dir, "package.json")
+	if err := os.Remove(packageJSONPath); !os.IsNotExist(err) {
+		require.NoError(t, err, "Failed to delete %s", packageJSONPath)
 	}
 	yarnLock := filepath.Join(dir, "yarn.lock")
 	if err := os.Remove(yarnLock); !os.IsNotExist(err) {
 		require.NoError(t, err, "Failed to delete %s", yarnLock)
 	}
 
-	err = integration.RunCommand(t, "link @pulumi/pulumi",
```
Comment (Frassle) [inline pkg/codegen/nodejs/gen_program_test.go:43]: Not linking to the local pulumi anymore?

File: pkg/codegen/nodejs/gen_program_test.go [hunk 2]
```diff
@@ -29,41 +30,51 @@ func nodejsCheck(t *testing.T, path string, dependencies codegen.StringSet) {
...
-	}
-	result := make([]string, len(deps))
-	for i, d := range deps.SortedValues() {
+func nodejsPackages(t *testing.T, deps codegen.StringSet) map[string]string {
+	result := make(map[string]string, len(deps))
+	for _, d := range deps.SortedValues() {
 		r := fmt.Sprintf("@pulumi/%s", d)
 		v := func(s string) {
-			r = fmt.Sprintf("%s@%s", r, s)
+			result[r] = "^" + s
```
Comment (mikhailshilkov) [inline pkg/codegen/nodejs/gen_program_test.go:77]: Could you add variable names or comments to explain what's going on here? `v` and `s` aren't that helpful to me.

File: pkg/codegen/nodejs/gen_program_test.go [hunk 3]
```diff
@@ -29,41 +30,51 @@ func nodejsCheck(t *testing.T, path string, dependencies codegen.StringSet) {
...
-		err = integration.RunCommand(t, "yarn add and install",
-			[]string{ex, "add", pkg}, dir, &integration.ProgramTestOptions{})
-		require.NoError(t, err, "Could not install package: %q", pkg)
+	pkgInfo := npmPackage{
+		Dependencies: map[string]string{
+			"@pulumi/pulumi": "latest",
+		},
+		DevDependencies: map[string]string{
+			"@types/node": "latest",
+			"typescript":  "latest",
```
Comment (mikhailshilkov) [inline pkg/codegen/nodejs/gen_program_test.go:50]: Is it possible that those `latest` versions may produce flaky results over time?

File: pkg/codegen/nodejs/gen_program_test.go [hunk 4]
```diff
@@ -29,41 +30,51 @@ func nodejsCheck(t *testing.T, path string, dependencies codegen.StringSet) {
...
 
-	err = integration.RunCommand(t, "link @pulumi/pulumi",
-		[]string{ex, "link", "@pulumi/pulumi"},
-		dir, &integration.ProgramTestOptions{})
-	require.NoError(t, err, "Failed to link @pulumi/pulumi")
-	for _, pkg := range pkgs {
-		err = integration.RunCommand(t, "yarn add and install",
-			[]string{ex, "add", pkg}, dir, &integration.ProgramTestOptions{})
-		require.NoError(t, err, "Could not install package: %q", pkg)
+	pkgInfo := npmPackage{
```
Comment (mikhailshilkov) [inline pkg/codegen/nodejs/gen_program_test.go:44]: I'm curious - why have you decided to change from yarn add commands to writing the package file explicitly?

Comment (iwahbe) [reply pkg/codegen/nodejs/gen_program_test.go:44]: I think that the new form is more readable. We deal mostly with go types, so the code is cleaner to me. When I wrote the previous version, I didn't know that `npmPackage` existed.

File: pkg/codegen/nodejs/gen_program_test.go (hunk 3 above)
Comment (iwahbe) [reply pkg/codegen/nodejs/gen_program_test.go:50]: I can pin to the current version (latest at this moment). I don't think there will be meaningful flake from TS and node, but pinning will make sure.

**Review by mikhailshilkov — APPROVED** [review]
LGTM

**Pull request merged**